test/mailers/subscription_mailer_test.rb

```diff
@@ -0,0 +1,59 @@
+require 'katello_test_helper'
+
+module Katello
+  class SubscriptionMailertest < ActiveSupport::TestCase
+    def setup
+      @user = User.current = User.find(users('admin').id)
+
+      FactoryBot.create(:mail_notification,
+                        :name => 'subscriptions_expiring_soon',
+                        :description => 'A list of subscriptions expiring within 30 days',
+                        :mailer => 'Katello::SubscriptionMailer',
+                        :method => 'subscription_expiry',
+                        :subscription_type => 'report')
+
+      @user.mail_notifications << MailNotification[:subscriptions_expiring_soon]
+
+
+      @pool_not_expiring_soon = FactoryBot.create(:katello_pool,
```

Comment:
@jturel I wrote that to make sure subscriptions not expiring soon are not included in the email. Is that test not required?

Reply:
Gotcha! I don't see any assertion which says "this pool is not included in the email", though. If you removed this code I think all your tests would still pass.

Keep in mind that you're generating the email for the admin user and they can see all pools - including the fixtures that already exist in the database. With that in mind you can use one of those fixtures and assert that it's not present in the email body. This would probably work:

```
@pool_not_expiring_soon = katello_pools(:fixture_name)

refute_includes email.body.encoded, @pool_not_expiring_soon.subscription.name
```

Another way to write this test would be to assert that there is only one row in the email body. You can use the Nokogiri  gem for that: https://github.com/sparklemotion/nokogiri#synopsis

For either approach you could also `assert Katello::Pool.readable.size > 1` and we can conclude that the result is accurate

Reply:
@jturel  I didn't want to use a fixture because the tests might fail at a later date. 
`refute_includes email.body.encoded, @pool_not_expiring_soon.subscription.name` This would not be true anymore if the end date in the fixture is less than 30 days from the current date.